code/modules/projectiles/projectile.dm

```diff
@@ -822,6 +833,11 @@ So if we are on the 32th absolute pixel coordinate we are on tile 1, but if we a
 //----------------------------------------------------------
 
 /atom/proc/bullet_act(obj/projectile/proj)
+	if(((proj.ammo.flags_ammo_behavior & AMMO_PASS_THROUGH_MOVABLE) || (proj.ammo.flags_ammo_behavior & AMMO_PASS_THROUGH_TURF)) && HAS_TRAIT(proj, TRAIT_PROJ_HIT_SOMETHING))
```

Comment:
(proj.ammo.flags_ammo_behavior & AMMO_PASS_THROUGH_MOVABLE|AMMO_PASS_THROUGH_TURF) works. But uh, is it even possible to call bullet_act twice if you are not AMMO_PASS_THROUGH anyway? In that case the flags check is useless